week2/money-diary/main.js

```diff
@@ -0,0 +1,220 @@
+
+  currentBalance = 0;
+  currentWithdraw = 0;
+  currentDeposit = 0;
+
+  renderHistory();
+  addBalance();
+};
+
+const addBalance = () => {
+  const allWithdraw = document.querySelectorAll(".price_withdraw ");
+  const allDeposit = document.querySelectorAll(".price_deposit");
+
+  currentDeposit = 0;
+  currentWithdraw = 0;
+
+  allDeposit.forEach((price) => {
+    currentDeposit = currentDeposit + Number(price.innerHTML);
+  });
+
+  allWithdraw.forEach((price) => {
+    currentWithdraw = currentWithdraw + Number(price.innerHTML);
+  });
+
+  currentBalance = currentDeposit + currentWithdraw;
+
+  setBalance();
+};
+
+const setBalance = () => {
+  balance.innerHTML = "";
+  const balanceNum = document.createTextNode(currentBalance);
+  balance.appendChild(balanceNum);
+
+  deposit.innerHTML = "";
+  const totalDepositNum = document.createTextNode("+" + currentDeposit);
+  deposit.appendChild(totalDepositNum);
+
+  withdraw.innerText = "";
+  const totalWithdrawNum = document.createTextNode(currentWithdraw);
+  withdraw.appendChild(totalWithdrawNum);
+};
+
+// 모달
+
+const modalWrapper = document.querySelector(".modal");
+const modalCloseButton = document.querySelector(".modal_close_button");
+const modal_deposit_input = document.querySelector(".modal_deposit_input");
+const modal_withdraw_input = document.querySelector(".modal_withdraw_input");
+const deposit_category_option = document.querySelector("#select_deposit");
+const withdraw_category_option = document.querySelector("#select_withdraw");
+const addButton = document.getElementById("add_button");
+
+addButton.onclick = () => {
+  modalWrapper.style.display = "flex";
+};
+
+modalCloseButton.onclick = () => {
+  modalWrapper.style.display = "none";
+};
+
+let newCategory = document.getElementById("select_deposit");
+let tempType = "price_deposit";
+
+const checkOnly = (e) => {
+  if (e.target === modal_deposit_input) {
+    modal_deposit_input.checked = true;
+    modal_withdraw_input.checked = false;
+
+    deposit_category_option.style.display = "flex";
+    withdraw_category_option.style.display = "none";
+
+    tempType = "price_deposit";
+
+    newCategory = document.getElementById("select_deposit");
+  } else {
+    modal_deposit_input.checked = false;
+    modal_withdraw_input.checked = true;
+
+    deposit_category_option.style.display = "none";
+    withdraw_category_option.style.display = "flex";
+
+    tempType = "price_withdraw";
+
+    newCategory = document.getElementById("select_withdraw");
+  }
+};
+
+modal_deposit_input.addEventListener("click", checkOnly);
+modal_withdraw_input.addEventListener("click", checkOnly);
+
+const titleInput = document.querySelector(".input_title");
+const priceInput = document.querySelector(".input_price");
+
+const newHistory = () => {
+  let tempCategory = newCategory.value;
+  let tempTitle = titleInput.value;
+  let tempPrice = priceInput.value;
+
+  if (tempCategory === "" || tempTitle === "" || tempPrice === "") {
```

Comment:
```suggestion
  if (!tempCategory || !tempTitle || !tempPrice) {
```
빈 문자열은 falsy값이라서 이렇게도 가능하답니다!